pkg/kubectl/cmd/delete.go

```diff
@@ -167,6 +173,9 @@ func (o *DeleteOptions) Complete(f cmdutil.Factory, args []string, cmd *cobra.Co
 		o.WaitForDeletion = true
 		o.GracePeriod = 1
 	}
+	if b, err := cmd.Flags().GetBool("wait"); err == nil {
```

Comment:
If I understand correctly, there is no real case when `err != nil` (if there are no bugs in code)? If so, setting `o.WaitForDeletion = true` 2 lines above becomes useless with this change.

Reply:
It happens during bad delegation and this command is ripe with it